### PR TITLE
gnome-monitor-config: init at 0-unstable-2023-09-26

### PR DESCRIPTION
**Pull request by aveltras**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/desktops/gnome/default.nix [hunk 1]
```diff
@@ -236,6 +236,8 @@ lib.makeScope pkgs.newScope (self: with self; {
 
   gnome-flashback = callPackage ./misc/gnome-flashback { };
 
+  gnome-monitor-config = callPackage ./misc/gnome-monitor-config { };
```
Comment (jtojnar) [inline pkgs/desktops/gnome/default.nix:239]: New packages should go to `by-name`.

File: pkgs/desktops/gnome/misc/gnome-monitor-config/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+  ];
+
+  buildInputs = [
+    cairo
```
Comment (jtojnar) [inline pkgs/desktops/gnome/misc/gnome-monitor-config/default.nix:21]: Missing `glib` (do not rely on propagation).

File: pkgs/desktops/gnome/misc/gnome-monitor-config/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    mv src/gnome-monitor-config $out/bin
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A program to help manage Gnome monitor configuration";
```
Comment (jtojnar) [inline pkgs/desktops/gnome/misc/gnome-monitor-config/default.nix:32]: GNOME should be capitalized.

File: pkgs/desktops/gnome/misc/gnome-monitor-config/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    homepage = "https://github.com/jadahl/gnome-monitor-config";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ ];
```
Comment (jtojnar) [inline pkgs/desktops/gnome/misc/gnome-monitor-config/default.nix:35]: Missing maintainer.

**Comment by jtojnar** [comment]
Looks like there is also https://github.com/NixOS/nixpkgs/pull/290444

File: pkgs/desktops/gnome/misc/gnome-monitor-config/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{ lib, fetchFromGitHub, stdenv, meson, ninja, pkg-config, cairo }:
+
+stdenv.mkDerivation rec {
+  pname = "gnome-monitor-config";
+  version = "2023-09-26";
```
[the remark this answers is not in the document]
Comment (aveltras) [reply pkgs/desktops/gnome/misc/gnome-monitor-config/default.nix:5]: nevermind it's here https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

**Comment by aveltras** [comment]
> Looks like there is also #290444

I saw that one too late while opening my PR but proceeded anyway since it seems inactive.
@jtojnar Should be OK for the fixes now.

File: pkgs/by-name/gn/gnome-monitor-config/package.nix [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    runHook preInstall
+    mkdir -p $out/bin
+    mv src/gnome-monitor-config $out/bin
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A program to help manage GNOME monitor configuration";
+    homepage = "https://github.com/jadahl/gnome-monitor-config";
+    license = licenses.gpl2Plus;
```
Comment (Aleksanaa) [inline pkgs/by-name/gn/gnome-monitor-config/package.nix:43]: Please add `mainProgram`.

Comment (aveltras) [reply pkgs/by-name/gn/gnome-monitor-config/package.nix:43]: done

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**